File: GFG/Strings/checkana.py
def ana(s1,s2):
    n1=len(s1)
    n2=len(s2)
    m={}
    for i in range(n1):
        if s1[i] not in m:
            m[s1[i]]=1
        else:
            m[s1[i]]+=1
    for j in range(n2):
        m[s2[j]]-=1
    cnt=sum(m.values())
    
    return cnt==0

q1="listen"
q2="silent"
print(ana(q1,q2))

File: GFG/Strings/checkana.cpp
#include<iostream>
#include<string>
#include<unordered_map>
using namespace std;

bool ana(string s1, string s2){
    int n1=s1.size();
    int n2=s2.size();
    unordered_map<char,int> m;
    for(int i=0;i<n1;i++){
        if(m.find(s1[i])==m.end()){
            m[s1[i]]=1;
        }
        else{
            m[s1[i]]++;
        }
    }
    for(int j=0;j<n2;j++){
        if(m.find(s2[j])==m.end()){
            return false;
        }
        else{
            m[s2[j]]--;
        }
    }
    int cnt=0;
    for(auto i:m){
        cnt+=i.second;
    }
    if(cnt!=0){
        return false;
    }
    return true;
}

int main(){
    string q1="listen";
    string q2="silent";
    cout<<ana(q1,q2);
    return 0;
}